src/PrestaShopBundle/ApiPlatform/DomainSerializer.php

```diff
@@ -53,17 +53,33 @@ public function __construct(Traversable $denormalizers)
      */
     public function denormalize($data, string $type, string $format = null, array $context = []): mixed
     {
-        $action = $this->serializer->denormalize($data, $type, $format, $context);
+        $dataConstruct = [];
+
+        $reflectionClass = new \ReflectionClass($type);
+        if($reflectionClass->getConstructor()) {
+            $constructParameters = $reflectionClass->getConstructor()->getParameters();
+
+            foreach ($constructParameters as $constructParameter) {
+                if (isset($data[$constructParameter->getName()])) {
+                    $dataConstruct[$constructParameter->getName()] = $data[$constructParameter->getName()];
+                    unset($data[$constructParameter->getName()]);
```

Comment:
Why do you need to handle the constructor's parameters separately? The serializer can only handle the constructor parameters?
Does it create some bugs if the data is still present for setter assignation later?

Reply:
No It's because denormalize call itself setters and there is an exception when it try to call 'setCartRuleAction' because it want a CartRuleAction and we have to denormalize this type from array before